component/newstorage/couchdb/store.go

```diff
@@ -535,14 +552,39 @@ func (s *Store) getRevID(k string) (string, error) {
 		return "", err
 	}
 
-	revID, err := getValueFromRawDoc(rawDoc, revIDFieldKey)
+	revID, err := getStringValueFromRawDoc(rawDoc, revIDFieldKey)
 	if err != nil {
 		return "", fmt.Errorf("failed to get revision ID from the raw document: %w", err)
 	}
 
 	return revID, nil
 }
 
+// getRawDocs returns the raw documents from CouchDB using a bulk REST call.
+// If a document is not found, then the raw document will be nil. It is not considered an error.
+func (s *Store) getRawDocs(keys []string) ([]map[string]interface{}, error) {
+	bulkGetReferences := make([]kivik.BulkGetReference, len(keys))
+	for i, key := range keys {
+		bulkGetReferences[i].ID = key
+	}
+
+	rows, err := s.db.BulkGet(context.Background(), bulkGetReferences)
```

Comment:
Reminder that contexts need to be passed-in.

Reply:
Created a followup issue earlier for it: https://github.com/hyperledger/aries-framework-go-ext/issues/48

Reply:
My goal is to get the storage implementations done first so that I can update the aries-framework-go functionality to use the new storage interface. Then I'll go and make those improvements that can be done easily afterwards.